src/functions.date.js

```diff
@@ -108,28 +108,56 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function($this, ms) {
+_.duration = $.extend(function ($this, ms, terms) {
 	if (arguments.length === 1) {
 		[ms, $this] = [$this, null];
 	}
 
-	var count = ms || 0;
-	var unit = "ms";
+	if (ms===0 || terms === undefined) {
+		terms = 1;
+	}
+
+	let timeLeft = ms || 0;
+	let ret = [];
+	let unitsToUse = ["ms"];
 
+	// populate unitsToUse with all the terms that can fit within timeLeft
+	// in order of lowest to highest magnitude,
+	// so we have a list of consecutive unit terms to later reference
 	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
+		let count = _.msTo(nextUnit, timeLeft);
 
-		if (nextCount === 0) {
+		if (count === 0) {
 			break;
 		}
 
-		count = nextCount;
-		unit = nextUnit;
+		unitsToUse.push(nextUnit);
 	}
 
-	unit = count === 1 && unit !== "ms"? unit.slice(0, -1) : unit;
+	while (ret.length < terms && unitsToUse.length > 0) {
+		let unit = unitsToUse.pop();
+		let unitTime;
+
+		if (unit!=="ms") {
+			unitTime = _.msTo(unit, timeLeft);
+		}
+		else {
+			unitTime = timeLeft;
+		}
```

Comment:
Maybe a ternary here?

`let unitTime = unit === "ms"? timeLeft : _.msTo(unit, timeLeft);`

Also, I'm not sure `unitTime` is very descriptive, I had to follow the code to understand what it's supposed to be. However I don't have any suggestions for a better name right now.